docs/memory.md

```diff
@@ -0,0 +1,104 @@
+- Count - Total number of current objects in the heap.
+- Accumulator - Total number of objects in the heap since the last reset.
+- Class - An aggregate of the objects allocated to this class. Clicking the class takes you to the list of all instances of this class.
+
+## Instances of a Class
+Displays a list of all instances by their handle name. _TODO INSPECTING AN INSTANCE_.
+
+## Memory Actions
+
+### Liveness of the Memory Usage Overview Chart
+- Pause - Pause the memory usage overview chart to allow inspecting of of the data currently being plotted.  New memory data continues to receive new data, notice the Range Selector grows, to the right.
+- Resume - The memroy usage overview chart is live and displaying the current time and the latest memory data received.
+
+### Managing the Objects and Statistics in the Heap
+- Snapshot - Returns the list of all active classes in the heap.  The Accumulator column displays the number of allocated objects since the previous "Reset".
+- Reset - Zeroes out the Accumulator column in the Snapshot Classes table and refreshes the displayed data.
+- Filter - TODO
+- GC - Initiates a garbage collection.
+
+## Glossary of VM Terms
+To truly understand how DevTool works, you need to understand computer science concepts such as memory allocation, the heap, garbage collection, and memory leaks. This glossary contains brief definitions of some of the terms used by DevTool.
+- Garbage collection - (GC) is the process of searching the heap to locate, and reclaim, regions of “dead” memory—memory that is no longer being used by an application. This process allows the memory to be re-used and minimizes the risk of an application running out of memory, causing it to crash. Garbage collection is performed automatically by the Dart VM. In DevTool, you can perform garbage collection on demand by clicking the GC button.
+- Heap - Dart objects that are dynamically allocated live in a portion of memory called the heap. An object allocated from the heap is freed (eligible for garbage collection) when nothing points to it, or when the application terminates. When nothing points to an object, it is considered to be dead. When an object is pointed to by another object, it is live.
+- Isolates - Dart supports concurrent execution by way of isolates, which you can think of as processes without the overhead. Each isolate has its own memory and code, which can’t be affected by any other isolate. For more information, see The Event Loop and Dart.
+- Memory leak - A memory leak occurs when an object is live (meaning that another object points to it) but it is not being used (so it shouldn’t have any references from other objects). Such an object can’t be garbage collected, so it takes up space in the heap and contributes to memory fragmentation. Memory leaks put unnecessary pressure on the VM and can be difficult to debug.
+- Virtual machine (VM) - The Dart virtual machine is a piece of software that can directly execute Dart code.
```

Comment:
if we want to get technical it is now a bytecode VM as it consumes kernel files.

Reply:
that can directly execute => that directly executes

Reply:
This was text stolen from the observatory docs.

If we use bytecode we'll probably need to define that too.  Sham is there any doc we can link to for a good VM description or a description of the VM that I can steal?